txscript/engine.go

```diff
@@ -652,7 +808,8 @@ func NewEngine(scriptPubKey []byte, tx *wire.MsgTx, txIdx int, flags ScriptFlags
 	// allowing the clean stack flag without the P2SH flag would make it
 	// possible to have a situation where P2SH would not be a soft fork when
 	// it should be.
-	vm := Engine{flags: flags, sigCache: sigCache}
+	vm := Engine{flags: flags, sigCache: sigCache, hashCache: hashCache,
+		inputAmount: inputAmount}
 	if vm.hasFlag(ScriptVerifyCleanStack) && !vm.hasFlag(ScriptBip16) {
```

Comment:
Nice catch, fixed (with a correction in the conditional to make the `||` an `&&`).